pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -309,6 +307,36 @@ private void messageReceived(ConsumerImpl<T> consumer, Message<T> message) {
         }
     }
 
+    void notifyPendingBatchReceivedCallBack() {
+        OpBatchReceive<T> opBatchReceive = pendingBatchReceives.poll();
+        if (opBatchReceive == null || opBatchReceive.future == null) {
+            return;
+        }
+        notifyPendingBatchReceivedCallBack(opBatchReceive);
+    }
+
+    void notifyPendingBatchReceivedCallBack(OpBatchReceive<T> opBatchReceive) {
+        MessagesImpl<T> messages = new MessagesImpl<>(batchReceivePolicy.getMaxNumMessages(),
+                batchReceivePolicy.getMaxNumBytes());
+        Message<T> msgPeeked = incomingMessages.peek();
+        while (msgPeeked != null && messages.canAdd(msgPeeked)) {
+            Message<T> msg = null;
+            try {
+                msg = incomingMessages.poll(0L, TimeUnit.MILLISECONDS);
+            } catch (InterruptedException e) {
+                // ignore
```

Comment:
@codelipenghui Can we just use `poll()`?

Reply:
yes fix it